dolphinscheduler-service/src/main/java/org/apache/dolphinscheduler/service/process/ProcessServiceImpl.java

```diff
@@ -793,6 +793,14 @@ private Boolean checkCmdParam(Command command, Map<String, String> cmdParam) {
                 break;
             case RECOVER_TOLERANCE_FAULT_PROCESS:
                 // recover tolerance fault process
+                // If the workflow instance is in ready state, we will change to running, this can avoid the workflow
+                // instance
+                // status is not correct with taskInsatnce status
+                if (processInstance.getState() == WorkflowExecutionStatus.READY_PAUSE
+                        || processInstance.getState() == WorkflowExecutionStatus.READY_STOP) {
+                    // todo: If we handle the ready state in WorkflowExecuteRunnable then we can remove below code
```

Comment:
May I ask why we did not handle the ready state directly, this approach seems to be a workaround

Reply:
This PR is only a simple bug fix, we need to handle the ready status WorkflowExecuteRunnable failover.